stl/inc/format

```diff
@@ -0,0 +1,363 @@
+        ++_Begin;
+        if(_Begin != _End) {
+            _Begin = _Parse_arg_id(_Begin, _End, _Width_adapter(_Callbacks));
+        }
+        if(_Begin == _End || *_Begin != '}') {
+            throw _STD format_error("Invalid format string");
+        }
+        ++_Begin;
+    }
+    return _Begin;
+
+}
+
+template <class _CharT>
+class basic_format_parse_context {
+public:
+    using char_type      = _CharT;
+    using const_iterator = typename basic_string_view<_CharT>::const_iterator;
+
+private:
+    basic_string_view<_CharT> _Format_string;
+    size_t _Num_args;
+    // The standard says this is size_t, however we use ptrdiff_t to save some space
+    // by not having to store the indexing mode. Below is a more detailed explination
+    // of how this works.
+    ptrdiff_t _Next_arg_id = 0;
+
+
+public:
+    constexpr explicit basic_format_parse_context(basic_string_view<_CharT> _Fmt, size_t _Num_args = 0) noexcept
+        : _Format_string(_Fmt), _Num_args(_Num_args) {}
+    basic_format_parse_context(const basic_format_parse_context&) = delete;
+    basic_format_parse_context& operator=(const basic_format_parse_context&) = delete;
+
+    constexpr const_iterator begin() const noexcept {
+        return _Format_string.begin();
+    }
+    constexpr const_iterator end() const noexcept {
+        return _Format_string.end();
+    }
+    constexpr void advance_to(const_iterator _It) {
+        _Format_string.remove_prefix(_It - begin());
+    }
+
+
+    // While the standard presents an exposition only enum value for
+    // the indexing mode (manual, automatic, or unknown) we use _Next_arg_id to indicate it.
+    // _Next_arg_id == 0 means unknown
+    // _Next_arg_id > 0 means automatic
+    // _Next_arg_id == -1 means manual
+    constexpr size_t next_arg_id() {
+        if (_Next_arg_id >= 0) {
+            return _Next_arg_id++;
+        }
+        throw _STD format_error("Can not switch from manual to automatic indexing");
+    }
+    constexpr void check_arg_id(size_t _Id) {
+        (void)_Id;
+        if (_Next_arg_id > 0) {
+            throw _STD format_error("Can not switch from automatic to manual indexing");
+        }
+        _Next_arg_id = -1;
+    }
+};
+
+
+template <class _CharT, class _Ty>
+struct _Named_arg {
+    const _CharT* _Name;
+    const _Ty& _Val;
+};
+
+template <class>
+constexpr bool _Is_named_arg_v = false;
+
+template <class _Ty, class _CharT>
+constexpr bool _Is_named_arg_v<_Named_arg<_CharT, _Ty>> = true;
+
+template <typename _Context>
+struct _Custom_value {
+    const void* _Value;
+    void (*_Format)(const void*, basic_format_parse_context<typename _Context::char_type>&, _Context&);
+};
+template <typename _Context>
+using _Format_arg_value =
+    _STD variant<monostate, int, unsigned, long long, unsigned long long, bool, typename _Context::char_type, float, double,
+        const void*, _STD basic_string_view<typename _Context::char_type>, _Custom_value<_Context>>;

```

Comment:
Missing `long double` and `const typename _Context::char_type*`. `_Custom_value<_Context>` should be (`typename basic_format_arg<_Context>::`)`handle`.

The difference is observable by `visit_format_arg`.